lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -254,6 +261,43 @@ def healthcheck!
       end
     end
 
+    def elasticsearch?(url)
+      begin
+        response = perform_request_to_url(url, :get, "/")
+      rescue ::LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError => e
+        return false if response.code == 401 || response.code == 403
+        raise e
+      end
+
+      version_info = LogStash::Json.load(response.body)
+      return false if version_info['version'].nil?
+
+      version = Gem::Version.new(version_info["version"]['number'])
+      return false if version < Gem::Version.new('6.0.0')
+
+      if version >= Gem::Version.new('6.0.0') && version < Gem::Version.new('7.0.0')
+        return valid_tagline?(version_info)
+      elsif version >= Gem::Version.new('7.0.0') && version < Gem::Version.new('7.14.0')
```

Comment:
We could use Gem::Requirements here to avoid creating new objects everytime:

```ruby
     VERSION_6_TO_7 = Gem::Requirement.new([">= 6.0.0", "< 7.0.0"])
     VERSION_7_TO_7_14 = Gem::Requirement.new([">= 7.0.0", "< 7.14.0"])
```

And then:

```suggestion
      if VERSION_6_TO_7.satisfied_by?(version)
        return valid_tagline?(version_info)
      elsif VERSION_7_TO_7_14.satisfied_by?(version)
```

Reply:
Nice, thanks! Applied